internal/packager/extract.go

```diff
@@ -65,42 +72,49 @@ func extractImage(appname string) (string, func(), error) {
 	}
 	tempDir, err := ioutil.TempDir("", "dockerapp")
 	if err != nil {
-		return "", noop, errors.Wrap(err, "failed to create temporary directory")
+		return ExtractedApp{}, errors.Wrap(err, "failed to create temporary directory")
 	}
 	defer os.RemoveAll(tempDir)
 	err = Load(imagename, tempDir)
 	if err != nil {
 		if !strings.Contains(imagename, "/") {
-			return "", noop, fmt.Errorf("could not locate application in either filesystem or docker image")
+			return ExtractedApp{}, fmt.Errorf("could not locate application in either filesystem or docker image")
 		}
 		// Try to pull it
 		cmd := exec.Command("docker", "pull", imagename)
 		if err := cmd.Run(); err != nil {
-			return "", noop, fmt.Errorf("could not locate application in filesystem, docker image or registry")
+			return ExtractedApp{}, fmt.Errorf("could not locate application in filesystem, docker image or registry")
 		}
 		if err := Load(imagename, tempDir); err != nil {
-			return "", noop, errors.Wrap(err, "failed to load pulled image")
+			return ExtractedApp{}, errors.Wrap(err, "failed to load pulled image")
 		}
 	}
 	// this gave us a compressed app, run through extract again
-	return Extract(filepath.Join(tempDir, appname))
+	appname, cleanup, err := Extract(filepath.Join(tempDir, appname))
+	return ExtractedApp{"", appname, cleanup}, err
+}
+
+// Extract bounces to ExtractWithOrigin discarding original app name
```

Comment:
nit: The comment should not descibe what the function is doing (implementation) but what is its behaviour (so in this case, most likely closer to what `ExtractWithOrigin` is)